model/host/volume.go

```diff
@@ -96,22 +96,17 @@ func FindVolumesToDelete(expirationTime time.Time) ([]Volume, error) {
 		},
 		{
 			"$lookup": bson.M{
-				"from":         Collection,
-				"localField":   "_id",
-				"foreignField": "volumes.volume_id",
-				"as":           "hosts",
-			},
-		},
-		{
-			"$project": bson.M{
-				"hosts": bson.M{
-					"$filter": bson.M{
-						"input": "$hosts",
-						"as":    "host",
-						"cond":  bson.M{"$ne": []string{"$$host.status", evergreen.HostTerminated}},
+				"from": Collection,
+				"let":  bson.M{"host_id": "$host"},
+				"pipeline": bson.A{
```

Comment:
Not sure if it's any more efficient, but could you do a localField/foreignField on _id, then filter out with a $match afterwards? I think it will at least be simpler (unless there's an array field here, in which case disregard)